platform/linuxbsd/wayland/display_server_wayland.cpp

```diff
@@ -643,6 +644,39 @@ bool DisplayServerWayland::screen_is_kept_on() const {
 #endif
 }
 
+bool DisplayServerWayland::screen_is_hdr_supported(int p_screen) const {
+	// TODO(ArchercatNEO): the color management protocol defines ICC color profiles not HDR
```

Comment:
You may be able to implement this by looking for one of the following:
- An output that supports HDR10: `primaries_named = bt2020` and `transfer_function = st2084_pq`
- An output that supports scRGB: `primaries_named = srgb` and `transfer_function = ext_linear`

I'm not very certain about scRGB checking, so we could start with only supporting HDR10 displays and add support for other formats later. Those other formats could be detected by checking their characteristics. 

I'm not sure how well Wayland implementations will handle converting between Godot's HDR10/scRGB output to what the display actually supports. These are early days in HDR support for Wayland.